src/screens/DeleteObservation/DeleteObservation.js

```diff
@@ -1,84 +1,101 @@
-import React, { Component } from "react";
+import React, { useState } from "react";
+import PropTypes from "prop-types";
 import { View, KeyboardAvoidingView } from "react-native";
 import { Overlay, Button, Text, Input } from "react-native-elements";
+/* eslint-disable-next-line */
+import { ObservationService } from "api";
 import Popup from "./Popup";
 import { translate } from "../../i18n";
 import { styles } from "./styles";
 
-class DeleteObservation extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      modalVisible: false
-    };
+const service = new ObservationService();
 
-    this.onPopupPress = this.onPopupPress.bind(this);
-  }
+const DeleteObservation = props => {
+  const [modalVisible, setModalVisible] = useState(false);
+  const [deleteDescription, setDeleteDescription] = useState("none");
 
-  onPopupPress(event, index) {
+  const { navigation } = props;
+  const onPopupPress = (event, index) => {
     if (index !== 0) {
-      this.setModalVisible(true);
+      setModalVisible(true);
     }
-  }
+  };
 
-  setModalVisible(visible) {
-    this.setState({ modalVisible: visible });
-  }
+  const onBackPress = () => {
+    setModalVisible(false);
+  };
 
-  render() {
-    const { modalVisible } = this.state;
-    const onBackPress = () => {
-      this.setModalVisible(false);
-    };
-    const onDeletePress = () => {
-      this.setModalVisible(false);
-    };
+  const onDeletePress = () => {
+    console.log(navigation.state.params.ObservationItem.id);
 
-    return (
-      <View>
-        <Overlay
-          isVisible={modalVisible}
-          windowBackgroundColor="rgba(0, 0, 0, .5)"
-        >
-          <KeyboardAvoidingView style={styles.container} enabled>
-            <Text h4 style={styles.headerText}>
-              {translate("deleteObservation.deleteObservationHeader")}
-            </Text>
-            <Text style={styles.hintText}>
-              {translate("deleteObservation.deleteObservationReason")}
-            </Text>
-            <Input
-              label={translate("deleteObservation.deleteObservationInputLabel")}
-              textContentType="emailAddress"
-              containerStyle={styles.input}
+    service
+      .deleteObservation(
+        navigation.state.params.ObservationItem.id,
+        deleteDescription
+      )
+      .then(response => {
+        console.log(response);
+        setModalVisible(false);
+        navigation.navigate("Observations");
+      })
+      .catch(err => {
+        console.log(err);
```

Comment:
We have to throw the error to show a message

Reply:
added alert to show message